### PR TITLE
Add test_cat_empty to allowed tests.

### PR DESCRIPTION
**Pull request by ailzhang**

### CODE REVIEW

**Comment by ailzhang** [comment]
test failure is real. Will check.

File: torch_xla/csrc/tensor_methods.cpp [hunk 1]
```diff
@@ -621,13 +621,24 @@ XLATensor XLATensor::cast(const XLATensor& input, at::ScalarType dtype) {
 XLATensor XLATensor::cat(tensorflow::gtl::ArraySlice<const XLATensor> tensors,
                          xla::int64 dim) {
   XLA_CHECK_GT(tensors.size(), 0);
+  xla::Shape template_shape;
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: What error are we trying to address here?
Can you post which case within that test fails?

Comment (ailzhang) [reply torch_xla/csrc/tensor_methods.cpp]: `cat` has a special logic when to throw errors... The test failure is still valid. I'll send a commit later.

File: torch_xla/csrc/tensor_methods.cpp [hunk 2]
```diff
@@ -620,14 +620,31 @@ XLATensor XLATensor::cast(const XLATensor& input, at::ScalarType dtype) {
...
+    xla::Shape tensor_shape = tensors[i].shape();
+    if (tensor_shape.rank() == 1 && tensor_shape.dimensions()[0] == 0) {
+      continue;
     }
+    dim = XlaHelpers::GetCanonicalDimensionIndex(dim, tensor_shape.rank());
+    tensor_shape.DeleteDimension(dim);
+    shapes.push_back(tensor_shape);
+    values.push_back(tensors[i].GetIrValue());
+  }
+  if (values.size() == 0) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: `values.empty()`

File: torch_xla/csrc/tensor_methods.cpp [hunk 3]
```diff
@@ -620,14 +620,31 @@ XLATensor XLATensor::cast(const XLATensor& input, at::ScalarType dtype) {
...
+    return tensors[0];
+  }
+  for (size_t i = 1; i < shapes.size(); ++i) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: Why an extra tail loop here?
While you are pushing above, you can check.
```C++
if (!shapes.empty()) {
  XLA_CHECK(xla::ShapeUtil::Equal(shapes.back(), tensor_shape)) << shapes.back() << " vs " << tensor_shape;
}
shapes.push_back(tensor_shape);
```

Comment (ailzhang) [reply torch_xla/csrc/tensor_methods.cpp]: Nice catch! Updating...

File: torch_xla/csrc/tensor_methods.cpp [hunk 4]
```diff
@@ -620,14 +620,29 @@ XLATensor XLATensor::cast(const XLATensor& input, at::ScalarType dtype) {
...
-      values.push_back(tensor.GetIrValue());
+  std::vector<xla::Shape> shapes;
+  for (size_t i = 0; i < tensors.size(); ++i) {
+    xla::Shape tensor_shape = tensors[i].shape();
+    if (tensor_shape.rank() == 1 && tensor_shape.dimensions()[0] == 0) {
+      continue;
     }
+    dim = XlaHelpers::GetCanonicalDimensionIndex(dim, tensor_shape.rank());
+    tensor_shape.DeleteDimension(dim);
+    XLA_CHECK(xla::ShapeUtil::Equal(shapes.back(), tensor_shape));
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: You need to check whether shapes is empty before testing ...

**Pull request merged**